src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -415,12 +413,16 @@ private static KeyExchangeProperties chooseCipherSuite(
             }
 
             List<CipherSuite> legacySuites = new LinkedList<>();
+            boolean CSFound = false;
```

Comment:
This variable may be not necessary if the debug log has been broken down into multiple place.

Reply:
There are two failing scenarios, one is "no common cipher suite" between the client and server and the other is the failure of key exchange after the server finds the shared cipher suites with the client. We are using the flag to give the developer a more specific error message.

Reply:
I see, but could you make it without define a variable that is purely for debug?